File: tournaments/cyclicSequence/cyclicSequence.js
function cyclicSequence(sequence) {
    let a = 0;
    let b = -1;

    for (let i = 1; i in sequence; ++i) {
        if (sequence[i - 1] === sequence[i]) {
            return false;
        }

        if (sequence[i - 1] > sequence[i]) {
            if (a) {
                return false;
            }

            a = 1;
            b = i;
        }
    }
    return ~b && sequence[0] > sequence[sequence.length - 1] || !~b;
}

File: tournaments/cyclicSequence/cyclicSequence.cpp
bool cyclicSequence(std::vector<int> sequence) {

  bool found = false;
  int st = -1;
  for (size_t i = 1; i < sequence.size(); i++) {
    if (sequence[i - 1] == sequence[i]) {
      return false;
    }
    if (sequence[i - 1] > sequence[i]) {
      if (found) {
        return false;
      }
      found = true;
      st = static_cast<int>(i);
    }
  }
  return st != -1 && sequence[0] > sequence.back() || st == -1;
}
